src/utils/codeowners.ts

```diff
@@ -63,17 +64,66 @@ const parseCodeOwner = (filePath: string, codeOwnerContent: string): ownerRule[]
   return filteredRules.map((rule) => ({ filePath, ...createMatcherCodeownersRule(filePath, rule) }));
 };
 
+const translateGlob = (glob: string) => {
+  const parsedGlob = parseGlob(glob);
+
+  // These controls are based on the Github CODEOWNERS syntax documentation
+  // https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners#codeowners-syntax
+  // as well as the gitignore pattern format which it extends
+  // https://git-scm.com/docs/gitignore#_pattern_format
+
+  if (parsedGlob.is.negated) {
+    throw new Error('Rule cannot contain negations');
+  }
+  if (parsedGlob.is.braces) {
+    throw new Error('Rule cannot contain braces');
+  }
+  if (glob.startsWith('\\#')) {
+    throw new Error('Rule cannot start with escaped #');
+  }
+  if (parsedGlob.is.glob && /\[(?:.-.)+\]/.test(parsedGlob.glob)) {
+    throw new Error('Rule cannot contain brackets');
+  }
```

Comment:
I often prefer to have separation of concerns. here is clear that there are 2 types of glob treatment going on: 
1) validation (is this glob that we got a glob we can work with or not?
2) is a valid glob but might need some treatment. 

I would group all these if's in a single function that is `validateGlob`

Reply:
True that. I split it out to a function called `isValidCodeownersGlob`. I also removed `is-valid-glob` since it apparently pretty much checks if it's a non-empty string: 

https://github.com/micromatch/is-valid-glob/blob/c18c82999df11133c5946f9a21ef74a0ebf88334/index.js#L4-L6